### PR TITLE
 Add ability to mark methods as ignored (not loaded)

### PR DESCRIPTION
**Pull request by NathanJPhillips**
This allows for methods such as default constructors that appear in overlay classes but have no effect and shouldn't be analyzed.
There may well be other use-cases for which this would be useful.
You may well want to ignore whitespace when diffing the second commit.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_parse_tree.h [hunk 1]
```diff
@@ -84,6 +84,11 @@ class java_bytecode_parse_treet
       is_private(false), is_static(false), is_final(false)
     {
     }
+
+    bool has_annotation(irep_idt annotation_id) const
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_parse_tree.h]: Pass by const reference, not by value.

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: ff555cc1cf86a9a8f5851c0ba891d0e30328936e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77983105
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 4a12a29d94cc20f91e87b6ad04574aabe20b8baa).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77985743

**Review by smowton — APPROVED** [review]

**Comment by allredj** [comment]
@NathanJPhillips I can't find the tests for this feature. Are they in another PR?

**Pull request merged**